### PR TITLE
feat(media): add ice failure groupings for CA

### PR DESCRIPTION
**Pull request by torgeadelin**
# COMPLETES https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-480923

## This pull request addresses
- categorize ice failures with more granular errors

## by making the following changes
- refactor
- add new client errors codes

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [x] Internal code refactor

## The following scenarios where tested
-unit test

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/internal-plugin-metrics/src/call-diagnostic/config.ts [hunk 1]
```diff
@@ -354,6 +362,26 @@ export const CLIENT_ERROR_CODE_TO_ERROR_PAYLOAD: Record<number, Partial<ClientEv
...
+    errorDescription: ERROR_DESCRIPTIONS.DTLS_HANDSHAKE_FAILED,
+    category: 'media',
+    fatal: true,
+  },
+  [ICE_FAILED_WITHOUT_TURN_TLS_CLIENT_CODE]: {
+    errorDescription: ERROR_DESCRIPTIONS.ICE_FAILED_WITHOUT_TURN_TLS,
+    category: 'media',
+    fatal: true,
+  },
+  [ICE_FAILED_WITHOUT_TURN_TLS_CLIENT_CODE]: {
```
Comment (marcin-bazyl) [inline packages/@webex/internal-plugin-metrics/src/call-diagnostic/config.ts:380]: this should be "with" not "without". I see you have a unit test for this, so not sure why it didn't fail, can you check that too?

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.util.ts [hunk 2]
```diff
@@ -538,4 +546,21 @@ describe('internal-plugin-metrics', () => {
...
     });
   });
+
+  describe('generateClientErrorCodeForIceFailure', () => {
+    [
+     { signalingState: 'have-local-offer', iceConnectionState: 'connected', turnServerUsed: true, errorCode: MISSING_ROAP_ANSWER_CLIENT_CODE},
+     { signalingState: 'stable', iceConnectionState: 'connected', turnServerUsed: true, errorCode: DTLS_HANDSHAKE_FAILED_CLIENT_CODE},
+     { signalingState: 'stable', iceConnectionState: 'failed', turnServerUsed: true, errorCode: ICE_FAILED_WITH_TURN_TLS_CLIENT_CODE},
+     { signalingState: 'stable', iceConnectionState: 'failed', turnServerUsed: false, errorCode: ICE_FAILED_WITHOUT_TURN_TLS_CLIENT_CODE},
+     { signalingState: 'failed', iceConnectionState: 'connected', turnServerUsed: false, errorCode: ICE_FAILURE_CLIENT_CODE},
```
Comment (marcin-bazyl) [inline packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.util.ts:556]: this case is not realistic, I don't think you can have iceConnectionState "connected" with signalingState "failed"

**Review by marcin-bazyl — CHANGES_REQUESTED** [review]

File: packages/@webex/internal-plugin-metrics/src/call-diagnostic/config.ts (hunk 1 above)
Comment (torgeadelin) [reply packages/@webex/internal-plugin-metrics/src/call-diagnostic/config.ts]: addressed

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.util.ts (hunk 2 above)
Comment (torgeadelin) [reply packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.util.ts]: addressed

File: packages/@webex/test-helper-chai/src/index.js [hunk 3]
```diff
@@ -8,6 +8,9 @@ const sinon = require('sinon');
 const registerAssertions = require('./assertions');
 
 chai.use(registerAssertions);
+
+chai.config.truncateThreshold = 0;
```
Comment (marcin-bazyl) [inline packages/@webex/test-helper-chai/src/index.js:12]: ah, thanks for this! I didn't know it was configurable, it's been so annoying when it got truncated

**Review by marcin-bazyl — APPROVED** [review]

File: packages/@webex/test-helper-chai/src/index.js (hunk 3 above)
Comment (torgeadelin) [reply packages/@webex/test-helper-chai/src/index.js:12]: yeah, it took a while to figure out but i was so glad i was able to fix it.
the config truncates it by default...

**Pull request merged**